File: examples/run_monitor.py
# run_monitor.py
# Runs the Diario Oficial Chile Monitor Actor and prints new/updated gazette entries.
import os
from apify_client import ApifyClient

# export APIFY_TOKEN=your_apify_api_token before running this script
client = ApifyClient(os.environ["APIFY_TOKEN"])

# Realistic minimal input: today's Normas Generales section, delta mode on.
run_input = {
    "sections": ["normas_generales"],
    "maxItems": 100,
    "onlyNew": True,
    "eventTypes": ["NEW_LISTING", "UPDATED"],
}

print("Starting diario-oficial-cl-monitor run...")
run = client.actor("qfBeEKuLfYUw9UOuW").call(run_input=run_input)

print(f'Run {run["id"]} finished with status "{run["status"]}".')

# Fetch the resulting dataset items (gazette entries) once the run completes.
dataset_items = client.dataset(run["defaultDatasetId"]).list_items().items

print(f"Fetched {len(dataset_items)} gazette publication(s):")
for item in dataset_items:
    print(f"- [{item['event_type']}] {item['ministerio']}: {item['descripcion']}")
    print(f"  PDF: {item['pdfUrl']}")

File: examples/run-monitor.js
// run-monitor.js
// Runs the Diario Oficial Chile Monitor Actor and prints new/updated gazette entries.
const { ApifyClient } = require('apify-client');

const client = new ApifyClient({
    token: process.env.APIFY_TOKEN, // export APIFY_TOKEN=your_apify_api_token first
});

async function main() {
    // Realistic minimal input: today's Normas Generales section, delta mode on.
    const input = {
        sections: ['normas_generales'],
        maxItems: 100,
        onlyNew: true,
        eventTypes: ['NEW_LISTING', 'UPDATED'],
    };

    console.log('Starting diario-oficial-cl-monitor run...');
    const run = await client.actor('qfBeEKuLfYUw9UOuW').call(input);

    console.log(`Run ${run.id} finished with status "${run.status}".`);

    // Fetch the resulting dataset items (gazette entries) once the run completes.
    const { items } = await client.dataset(run.defaultDatasetId).listItems();

    console.log(`Fetched ${items.length} gazette publication(s):`);
    for (const item of items) {
        console.log(`- [${item.event_type}] ${item.ministerio}: ${item.descripcion}`);
        console.log(`  PDF: ${item.pdfUrl}`);
    }
}

main().catch((err) => {
    console.error('Run failed:', err.message);
    process.exit(1);
});
